### ISSUE TITLE
Unescaped regex-meta-character in AbstractFormDateSelect

### ISSUE THREAD

**Issue by strouble** [laminas/laminas-form#160]
I think the regex-meta-character "-" (dash) in the date/time pattern should be escaped:
https://github.com/laminas/laminas-form/blob/beee25df568f1140d5553eabe51f97c6643f0464/src/View/Helper/AbstractFormDateSelect.php#L78

It should be:
```
"/([ \-,.\/]*(?:'[a-zA-Z]+')*[ \-,.\/]+)/"
```

Otherwise dates in the format `dd-MM-yy` are not recognized.
For example, the `IntlDateFormatter` returns for the `es_CL` locale that pattern.

**Comment by froschdesign** [comment]
@strouble 
Thanks for reporting! 👍 

> Otherwise dates in the format `dd-MM-yy` are not recognized.

Correct, this format can not be parsed. Here is a short example:

```php
// Create renderer
/** @var Laminas\View\Renderer\PhpRenderer|Laminas\Form\View\HelperTrait $renderer */
$renderer = new Laminas\View\Renderer\PhpRenderer();
$renderer->getHelperPluginManager()->configure(
    (new Laminas\Form\ConfigProvider())->getViewHelperConfig()
);
$renderer->doctype(Laminas\View\Helper\Doctype::HTML5);

// Create form element
$dateElement = new Laminas\Form\Element\DateSelect('test-date');

// Render form element
echo $renderer->formDateSelect(
    $dateElement,
    IntlDateFormatter::SHORT,
    'es_CL'
);
```

This produces:

> Warning: Undefined array key "month" in /…/laminas-form/src/View/Helper/FormDateSelect.php on line 75

Because the parsed pattern is:

```
array(1) {
  ["day"]=>
  string(8) "dd-MM-yy"
}
```

https://github.com/laminas/laminas-form/blob/beee25df568f1140d5553eabe51f97c6643f0464/src/View/Helper/AbstractFormDateSelect.php#L74-L98

> It should be:
> 
> ```
> "/([ \-,.\/]*(?:'[a-zA-Z]+')*[ \-,.\/]+)/"
> ```

Would you provide a pull request to fix the problem?

**Comment by Ocramius** [comment]
Handled in #184 by @mimmi20, and released in `3.4.1`, thanks everyone!
